src/controllers/project.js

```diff
@@ -1,9 +1,113 @@
 const { ObjectId } = require('mongodb');
 const elasticsearch = require('elasticsearch');
+const mongoose = require('mongoose');
 
 const Project = require('../models/project');
 const User = require('../models/user');
 
+const createProject = async (req, res) => {
+  try {
+    const { /* comment, review, //BUG not used */ project, user } = req.body;
+
+    const newProjects = new Project({
+      title: project.title,
+      description: project.description,
+      category: [],
+      donation: project.donation,
+      amount: project.amount,
+      collectedAmount: project.collectedAmount,
+      comments: [],
+      review: [],
+      owners: project.owners,
+      supporters: project.supporters,
+      user: mongoose.Types.ObjectId(user._id),
+    });
+
+    newProjects.category.push(...project.categories);
+    newProjects.save();
+    res.status(200).json('Project created successfully');
+  } catch (err) {
+    res.status(422).json('Could not create project');
+  }
+};
+
+const updateProject = async (req, res) => {
+  const { title, description, /* category //BUG not used */ donation } =
```

Comment:
here category not used
but categories do

you just should name the parameter categories